src/Skybrud.ImagePicker/PropertyEditors/ImagePicker/ValueConverters/ImageValueConverter.cs

```diff
@@ -69,57 +67,52 @@ public class ImageValueConverter : PropertyValueConverterBase {
 
             // Initialize a collection for the items
             List<object> items = new List<object>();
-            
+
             // Determine the item value type
             Type valueType = propertyType.DataType.ConfigurationAs<ImagePickerConfiguration>()?.ValueType;
 
-            foreach (Udi udi in udis)  {
-                
+            foreach (Udi udi in udis) {
+
+                var canGetPublishedSnapshot = _publishedSnapshotAccessor.TryGetPublishedSnapshot(out var publishedSnapshotAccessor);
+
+                if (!canGetPublishedSnapshot)
+                    continue;
+
                 // Look up the media
-                IPublishedContent media = _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(udi);
+                IPublishedContent media = publishedSnapshotAccessor.Media.GetById(udi);
                 if (media == null) continue;
 
                 // If the configuration doesn't specify a value type, we just create a new ImagePickerImage
                 if (valueType == null) {
-                    items.Add(new ImagePickerImage(media, config));
+                    items.Add(new Image(media, config));
                     continue;
                 }
 
                 // If the selected type has a constructor with an ImagePickerConfiguration as the second parameter, we choose that constructor
                 if (HasConstructor<IPublishedContent, ImagePickerConfiguration>(valueType)) {
-                    items.Add(Current.Factory.CreateInstance(valueType, media, config));
+                    items.Add(ActivatorUtilities.CreateInstance(_serviceProvider, valueType, media, config));
```

Comment:
Doesn't matter much, but might make the code easier to read, but Umbraco has an extension method so we can call `_serviceProvider.CreateInstance(valueType, media, config)` rather than `ActivatorUtilities.CreateInstance(_serviceProvider, valueType, media, config)`.